### PR TITLE
Remove problematic terms and facilitate default branch renaming

### PR DESCRIPTION
**Pull request by felixarntz**
## Summary

Addresses issue #3031

## Relevant technical choices

* The only changes here that go beyond changing wording were added in distinct commits - please review those separately.
* This PR leaves us with some problematic terms only in `composer.lock` and `package-lock.json`, as well as in a few other areas related to third-party code (e.g. stylelint and PHP-Scoper config).
    * `stylelint` already has improved language throughout its codebase too, so can update it. That comes with a bunch of side effects though, so should be done separately.

## Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: tests/phpunit/integration/Core/Util/ScopesTest.php [hunk 1]
```diff
@@ -96,7 +96,7 @@ public function is_satisfied_by_provider() {
 				),
 				true,
 			),
-			// webmaster is not satisfied by webmasters.readonly
+			// webmasters is not satisfied by webmasters.readonly
```
Comment (felixarntz) [inline tests/phpunit/integration/Core/Util/ScopesTest.php:99]: Unrelated change - just fixes a typo.

**Comment by tofumatt** [comment]
@googlebot I consent.

**Comment by tofumatt** [comment]
😆  Oops, I noticed the `zips` workflow still had the `master` branch listed and changed it, only to see that was intentional and from this branch. I'll change it back, but add a comment to note why it isn't changed, as that wasn't clear to me.

I venture they'll eventually change that too, hopefully, but we'll see.

**Review by tofumatt — APPROVED** [review]
Awesome 👍🏻

File: tests/bin/install-wp-tests.sh [hunk 2]
```diff
@@ -121,6 +119,9 @@ install_test_suite() {
 		sed $ioption "s/yourusernamehere/$DB_USER/" "$WP_TESTS_DIR"/wp-tests-config.php
 		sed $ioption "s/yourpasswordhere/$DB_PASS/" "$WP_TESTS_DIR"/wp-tests-config.php
 		sed $ioption "s|localhost|${DB_HOST}|" "$WP_TESTS_DIR"/wp-tests-config.php
+
+		# Enforce mysqli.
+		echo "define( 'WP_USE_EXT_MYSQL', false );" >> "$WP_TESTS_DIR"/wp-tests-config.php
```
Comment (tofumatt) [inline tests/bin/install-wp-tests.sh:124]: Just noting, for any future reviewers/on-lookers, that this change removes the link to a `master` repo URL and that the file being downloaded does what this new code does. If anything, this is probably more stable in case that repo is removed 🙂

Comment (aaemnnosttv) [reply tests/bin/install-wp-tests.sh:124]: @tofumatt @felixarntz I don't think we should bother making customizations to this file. This script is essentially an [external dependency](https://github.com/wp-cli/scaffold-command/blob/479cd4f10f638d7a46a5b4a60c80bd4b4a6bd5e2/templates/install-wp-tests.sh) that we mirror in our source because it's a shell script.

If anything, this patch should be an upstream contribution 🙂 

In the future, we should be able to update this by replacing this file with the newest version from upstream without requiring manual updates afterwards.
